to.etc.domui/src/main/java/to/etc/domui/dom/html/Page.java

```diff
@@ -1033,6 +1035,14 @@ void registerJavascriptStateChanged(@NonNull NodeBase nodeBase) {
 	 */
 	@Nullable
 	public StringBuilder internalFlushJavascriptStateChanges() throws Exception {
+		if(canLeaveCurrentPage(false)) {
+			if(m_rootContent instanceof IPageWithNavigationCheck) {
+				m_rootContent.appendJavascript("window.onbeforeunload = null;");
+			}
+		}else {
+			//seems to be hardcoded message in browsers, so text that we return here is ignored.
+			m_rootContent.appendJavascript("window.onbeforeunload = function() { return \"You have unsaved changes on page.\"};");
```

Comment:
That javascript should be a WebUI function so that we can easily change it. It also seems incomplete: the web states that you need to do:
```
window.addEventListener('beforeunload', (event) => {
            event.preventDefault();
           // Google Chrome requires returnValue to be set.
            event.returnValue = '';
        });
```
Returning a user message is not supported anymore for security.

Reply:
as agreed, made it to use WebUI functions.